keanu-project/src/test/java/io/improbable/keanu/vertices/intgr/IntegerVertexTest.java

```diff
@@ -66,4 +66,35 @@ public void doesLambda() {
         assertEquals(result.getValue().scalar(), expected);
     }
 
+    @Test
+    public void canObserveArrayOfValues() {
+        IntegerVertex binomialVertex = new BinomialVertex(0.5, 20);
+        int[] observation = new int[]{1, 2, 3};
+        binomialVertex.observe(observation);
+        assertArrayEquals(observation, binomialVertex.getValue().asFlatIntegerArray());
+    }
+
+    @Test
+    public void canSetAndCascadeArrayOfValues() {
+        IntegerVertex binomialVertex = new BinomialVertex(0.5, 20);
+        int[] values = new int[]{1, 2, 3};
+        binomialVertex.setAndCascade(values);
+        assertArrayEquals(values, binomialVertex.getValue().asFlatIntegerArray());
+    }
+
+    @Test
+    public void canSetValueAsArrayOfValues() {
+        IntegerVertex binomialVertex = new BinomialVertex(0.5, 20);
+        int[] values = new int[]{1, 2, 3};
+        binomialVertex.setAndCascade(values);
+        assertArrayEquals(values, binomialVertex.getValue().asFlatIntegerArray());
+    }
+
+    @Test
+    public void canSetValueAsScalarOnNonScalarVertex() {
+        IntegerVertex binomialVertex = new BinomialVertex(new int[]{2, 1}, 0.5, 20);
+        binomialVertex.setValue(2);
+        assertArrayEquals(new int[]{2}, binomialVertex.getValue().asFlatIntegerArray());
+    }
+
 }
```

Comment:
add a `canSetAndCascadeAsScalarOnNonScalarVertex` method (as in the `DoubleVertexTest`)?

Reply:
Yup. Good call.